### PR TITLE
Fix Okta auth to allow group names containing slashes

### PR DESCRIPTION
**Pull request by kalafut**
This PR also adds CollectKeysPrefix which allows a more memory efficient
key scan for those cases where the result is immediately filtered by
prefix.

### CODE REVIEW

File: sdk/logical/storage.go [hunk 1]
```diff
@@ -86,17 +86,24 @@ func ScanView(ctx context.Context, view ClearableView, cb func(path string)) err
 
 // CollectKeys is used to collect all the keys in a view
 func CollectKeys(ctx context.Context, view ClearableView) ([]string, error) {
-	// Accumulate the keys
-	var existing []string
+	return CollectKeysPrefix(ctx, view, "")
+}
+
+// CollectKeysPrefix is used to collect all the keys in a view with a given prefix string
+func CollectKeysPrefix(ctx context.Context, view ClearableView, prefix string) ([]string, error) {
```
Comment (briankassouf) [inline sdk/logical/storage.go]: Nit: how about `CollectKeysWithPrefix`?

File: builtin/credential/ldap/path_users.go [hunk 2]
```diff
@@ -148,18 +148,14 @@ func (b *backend) pathUserWrite(ctx context.Context, req *logical.Request, d *fr
...
+	keys, err := logical.CollectKeysPrefix(ctx, req.Storage, "user/")
 	if err != nil {
 		return nil, err
 	}
-	retKeys := make([]string, 0)
-	for _, key := range keys {
-		if strings.HasPrefix(key, "user/") && !strings.HasPrefix(key, "/") {
-			retKeys = append(retKeys, strings.TrimPrefix(key, "user/"))
-		}
+	for i := range keys {
```
Comment (briankassouf) [inline builtin/credential/ldap/path_users.go]: Could we remove the retKeys field in the file above too?

File: sdk/logical/storage.go (hunk 1 above)
Comment (kalafut) [reply sdk/logical/storage.go]: yep

File: builtin/credential/ldap/path_users.go (hunk 2 above)
Comment (kalafut) [reply builtin/credential/ldap/path_users.go]: Ah yeah, missed that one.

**Pull request merged**